pycalphad/core/starting_point.py

```diff
@@ -77,23 +51,39 @@ def starting_point(conditions, state_variables, phase_records, grid):
-             'X':      (conds_as_strings + ['vertex', 'component'],
-                        np.empty(grid_shape + (len(nonvacant_elements)+1, len(nonvacant_elements),))),
-             'Y':      (conds_as_strings + ['vertex', 'internal_dof'],
-                        np.empty(grid_shape + (len(nonvacant_elements)+1, maximum_internal_dof,))),
-             'Phase':  (conds_as_strings + ['vertex'],
-                        np.empty(grid_shape + (len(nonvacant_elements)+1,), dtype='U%s' % max_phase_name_len)),
-             'points': (conds_as_strings + ['vertex'],
-                        np.empty(grid_shape + (len(nonvacant_elements)+1,), dtype=np.int32))
-             },
-             coords=coord_dict, attrs={'engine': 'pycalphad %s' % pycalphad_version})
+    result = LightDataset(
+        {'NP':     (conds_as_strings + ['vertex'], np.empty(grid_shape + (max_phases,))),
+         'GM':     (conds_as_strings, np.empty(grid_shape)),
+         'MU':     (conds_as_strings + ['component'], np.empty(grid_shape + (len(nonvacant_elements),))),
+         'X':      (conds_as_strings + ['vertex', 'component'],
+                    np.empty(grid_shape + (max_phases, len(nonvacant_elements),))),
+         'Y':      (conds_as_strings + ['vertex', 'internal_dof'],
+                    np.empty(grid_shape + (max_phases, maximum_internal_dof,))),
+         'Phase':  (conds_as_strings + ['vertex'],
+                    np.empty(grid_shape + (max_phases,), dtype='U%s' % max_phase_name_len)),
+         'points': (conds_as_strings + ['vertex'],
+                    np.empty(grid_shape + (max_phases,), dtype=np.int32))
+         }, coords=coord_dict, attrs={'engine': 'pycalphad %s' % pycalphad_version})
+
+    if given_starting_point is None:
         result = lower_convex_hull(grid, state_variables, result)
     else:
-        raise NotImplementedError('Conditions not yet supported')
+        out_energy = np.zeros((len(given_starting_point), 1))
+        out_moles = np.zeros((len(given_starting_point), 1, len(nonvacant_elements)))
+        for phase_idx, (phase_name, phase_dof) in enumerate(given_starting_point):
+            phase_dof_without_statevars = phase_dof[len(state_variables):]
+            result['NP'][..., phase_idx] = 1./len(given_starting_point)
+            phase_records[phase_name].obj(out_energy[phase_idx], np.atleast_2d(phase_dof))
+            for comp_idx in range(len(nonvacant_elements)):
+                phase_records[phase_name].mass_obj(out_moles[phase_idx, :, comp_idx], np.atleast_2d(phase_dof), comp_idx)
+            result['Phase'][..., phase_idx] = phase_name
+            result['X'][..., phase_idx, :] = out_moles[phase_idx, 0]
+            result['Y'][..., phase_idx, :len(phase_dof_without_statevars)] = phase_dof_without_statevars
+            result['Y'][..., phase_idx, len(phase_dof_without_statevars):] = np.nan
+            out_energy[:] = 0
```

Comment:
If `len(given_starting_point) < max_phases` we should set the unused parts of the empty array to `np.nan`